interfaces/builtin/u2f_devices.go

```diff
@@ -162,6 +162,11 @@ var u2fDevices = []u2fDevice{
 		VendorIDPattern:  "16d0",
 		ProductIDPattern: "0e90",
 	},
+	{
+		Name:             "Flipper Devices Inc. U2F Token",
+		VendorIDPattern:  "0483",
```

Comment:
Thanks, that looks great! Do you have a link to some upstream docs around the flipper (cool device) that confirms that this is the right usb product id?

Reply:
0x0483 is 1155 which belongs to STMicroelectronics as per USB.org vendor ids list from https://www.usb.org/developers see https://www.usb.org/sites/default/files/vendor_ids063022_0.pdf

And 5741 product ID seems to be just a USB controller - as in it can be any USB device.

Flipper Devices should probably apply for their own unique vendor-id https://www.usb.org/getting-vendor-id for their products? instead of reusing their component's IDs?

Reply:
I don't have docs but I have some logs 
dmesg : 
[  138.268754] usb 1-1: new full-speed USB device number 5 using xhci_hcd
[  138.421090] usb 1-1: New USB device found, idVendor=0483, idProduct=5741, bcdDevice= 1.00
[  138.421103] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[  138.421109] usb 1-1: Product: U2F Token
[  138.421112] usb 1-1: Manufacturer: Flipper Devices Inc.
[  138.464097] hid: raw HID events driver (C) Jiri Kosina
[  138.469571] usbcore: registered new interface driver usbhid
[  138.469576] usbhid: USB HID core driver
[  138.486744] hid-generic 0003:0483:5741.0001: hiddev0,hidraw0: USB HID v1.00 Device [Flipper Devices Inc. U2F Token] on usb-0000:00:14.0-1/input0
lsusb:
Bus 001 Device 005: ID 0483:5741 STMicroelectronics U2F Token

Reply:
And at the line 62 and 63 of https://github.com/flipperdevices/flipperzero-firmware/blob/dev/firmware/targets/f7/furi_hal/furi_hal_usb_u2f.c

Reply:
Something like :
P: /devices/pci0000:00/0000:00:14.0/usb1/1-2
N: bus/usb/001/006
L: 0
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-2
E: SUBSYSTEM=usb
E: DEVNAME=/dev/bus/usb/001/006
E: DEVTYPE=usb_device
E: DRIVER=usb
E: PRODUCT=483/5741/100
E: TYPE=239/2/1
E: BUSNUM=001
E: DEVNUM=006
E: MAJOR=189
E: MINOR=5
E: USEC_INITIALIZED=1971671762
E: ID_VENDOR=Flipper_Devices_Inc.
E: ID_VENDOR_ENC=Flipper\x20Devices\x20Inc.
E: ID_VENDOR_ID=0483
E: ID_MODEL=U2F_Token
E: ID_MODEL_ENC=U2F\x20Token
E: ID_MODEL_ID=5741
E: ID_REVISION=0100
E: ID_SERIAL=Flipper_Devices_Inc._U2F_Token
E: ID_BUS=usb
E: ID_USB_INTERFACES=:030000:
E: ID_VENDOR_FROM_DATABASE=STMicroelectronics
E: ID_PATH=pci-0000:00:14.0-usb-0:2
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_2
E: TAGS=:snap_cups_ippeveprinter:snap_cups_cupsd:
E: CURRENT_TAGS=:snap_cups_ippeveprinter:snap_cups_cupsd:

Reply:
given above we cannot key onto numerical ID_VENDOR_ID / ID_MODEL_ID as both STMicroelectornics generic devices and Flipper U2F tokens have it identical. We must key onto string patterns ID_VENDOR=Flipper_Devices_Inc. and ID_MODEL=U2F_Token , somehow. so possibly additional code changes are needed here.

Reply:
Ok, I've searched where it is defined for the "Tomu board + chopstx U2F + SoloKeys" and "MIRKey" that have the same ID_VENDOR (STMicroelectornics) but I didn't find where to add ID_VENDOR and ID_MODEL.